hazelcast/src/main/java/com/hazelcast/query/impl/SkipIndexPredicate.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.query.impl;
+
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.impl.predicates.PredicateDataSerializerHook;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static com.hazelcast.internal.serialization.impl.FactoryIdHelper.PREDICATE_DS_FACTORY_ID;
+
+/**
+ * The {@link SkipIndexPredicate} is a predicate that prevents
+ * an index from being used on the wrapped predicate.
+ *
+ * It isn't exposed as user API; it will only be created when
+ * making use of index suppression option in the
+ * {@link com.hazelcast.query.SqlPredicate}
+ *
+ * SkipIndexPredicate also isn't send over the wire; so we don't
+ * need to worry about backward compatibility in future releases.
+ */
+public class SkipIndexPredicate implements Predicate, IdentifiedDataSerializable {
```

Comment:
why does it implement IDS? Is this enforced by some predicate-convention-tests?

Reply:
Yes. Initially I had no serialization logic and one of the compatibility tests started to complain about that.

Reply:
I believe you are referring to `DataSerializableConventionsTest`. There is a white-list in that test of other classes inheriting `Serializable` but not implementing `IdentifiedDataSerializable`:
https://github.com/hazelcast/hazelcast/blob/fa7054a4ce391404185fbed8361f20993b96b726/hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/DataSerializableConventionsTest.java#L67-L79

Please add this class to the whitelist and remove the `IdentifiedDataSerializable` interface.